contents/tutorials/api-feature-flags.md

```diff
@@ -0,0 +1,318 @@
+
+<MultiLanguage>
+
+```bash
+export POSTHOG_PERSONAL_API_KEY=<POSTHOG_PERSONAL_API_KEY>
+export POSTHOG_PROJECT_ID=<POSTHOG_PROJECT_ID>
+curl \
+  -H "Authorization: Bearer $POSTHOG_PERSONAL_API_KEY" \
+  https://app.posthog.com/api/projects/$POSTHOG_PROJECT_ID/feature_flags/<FLAG_ID>
+```
+
+```python
+import requests
+
+POSTHOG_PERSONAL_API_KEY = <POSTHOG_PERSONAL_API_KEY>
+
+headers = {"Authorization": f"Bearer {POSTHOG_PERSONAL_API_KEY}" }
+
+response = requests.get(
+  f"https://app.posthog.com/api/projects/<POSTHOG_PROJECT_ID>/feature_flags/<FLAG_ID>",
+  headers=headers
+).json()
+```
+
+</MultiLanguage>
+
+## Updating feature flag data
+
+We can use the same `/api/projects/:project_id/feature_flags/:id/` route with the `PATCH` method to update the data about the flag such as name, key, rollout percentage, or release conditions. For example, if we wanted to change the rollout percentage of our flag, we could make a request like this:
+
+<MultiLanguage>
+
+```bash
+export POSTHOG_PERSONAL_API_KEY=<POSTHOG_PERSONAL_API_KEY>
+export POSTHOG_PROJECT_ID=<POSTHOG_PROJECT_ID>
+curl \
+  -H "Authorization: Bearer $POSTHOG_PERSONAL_API_KEY" \
+  -H "Content-Type: application/json" \
+  -d '{"filters":{"groups":[{"properties":[],"rollout_percentage":0}]}}' \
+  -X PATCH https://app.posthog.com/api/projects/$POSTHOG_PROJECT_ID/feature_flags/<FLAG_ID>
+```
+
+```python
+POSTHOG_PERSONAL_API_KEY = "<POSTHOG_PERSONAL_API_KEY>"
+ROLLOUT_PERCENTAGE = 0
+
+headers = {"Authorization": f"Bearer {POSTHOG_PERSONAL_API_KEY}" }
+
+filters = {
+  "filters": {
+    "groups": [
+      {
+        "properties": [],
+        "rollout_percentage": ROLLOUT_PERCENTAGE
+      }
+    ]
+  }
+}
+
+response = requests.patch(
+  f"https://app.posthog.com/api/projects/<POSTHOG_PROJECT_ID>/feature_flags/<FLAG_ID>",
+  headers=headers,
+  json=filters
+).json()
+```
+
+</MultiLanguage>
+
+This filter object can include available user properties, cohorts, or group properties. You also can use the `PATCH` endpoint to change the name, key, filters, deleted status, active status, and more. Be careful though, it overwrites any of your current feature flag details. See the full details in [our API docs](/docs/api/feature-flags#patch-api-projects-project_id-feature_flags-id).
+
+## Combining our requests to update flags when evaluated
+
+To combine these, we will write a Python script that:
+
+1. evaluates our feature flag
+2. figures out the feature flag ID by finding the key in the "get all feature flags" request
+3. updates the feature flag to filter out the email of the user who saw the feature flag
+
+> The best way to do this is probably to set the release condition to not include a property you set on a person once they see the flag, but this is a tutorial about using the API so that is less fun.
```

Comment:
```suggestion
You can also do this by setting the release conditions to not include a property you set on a person that has seen the flag, if you prefer. 
```

I don't think this is worth calling out and, while I like saying we're focusing on the fun way to do it, a tutorial should usually detail the best way to do it.